example_flutter/lib/keyboard_test_page.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+
+class KeyboardTestPage extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return _KeyboardTestPageState();
+  }
+}
+
+class _KeyboardTestPageState extends State<KeyboardTestPage> {
+  final List<String> _messages = [];
+
+  final FocusNode _focusNode = FocusNode();
+  final ScrollController _scrollController = new ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    FocusScope.of(context).requestFocus(_focusNode);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: AppBar(
+          leading: new IconButton(
+              icon: new Icon(Icons.arrow_back),
+              onPressed: () {
+                Navigator.of(context).pop();
+              })),
+      body: new RawKeyboardListener(
+        focusNode: _focusNode,
+        onKey: onKeyEvent,
+        child: Container(
+          padding: EdgeInsets.symmetric(horizontal: 16.0),
+          child: SingleChildScrollView(
+              controller: _scrollController,
+              child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: _messages.map((m) => new Text(m)).toList())),
+        ),
+      ),
+    );
+  }
+
+  void onKeyEvent(RawKeyEvent event) async {
+    int keyCode;
+    bool isKeyDown;
```

Comment:
Prefer to declare variables as close to their first use as possible, so keyCode attached to the second switch below, and this attached to the first switch.